tests/test_experimental.py

```diff
@@ -1,14 +1,388 @@
+import functools
+import time
 from copy import deepcopy
 
 import numpy as np
 import pytest
 
 from iterative_ensemble_smoother import ESMDA
+from iterative_ensemble_smoother.esmda_inversion import normalize_alpha
 from iterative_ensemble_smoother.experimental import (
+    AdaptiveESMDA,
     ensemble_smoother_update_step_row_scaling,
+    groupby_indices,
 )
 
 
+@pytest.mark.parametrize("seed", range(25))
+def test_groupby_indices(seed):
```

Comment:
From my reading of this test, it does not test the "grouping" property of the function when multiple rows of `X` are identical.
Since we assert that `union_idx == set(range(rows))` we I think assert that `X` has no duplicate rows.
Am I understanding this correctly?